From f549d6146597f7f9912ff7560f4e04bd17656729 Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 11 Sep 2025 18:13:27 +0800
Subject: [PATCH 23/32] dts: ast2700-evb: update memory nodes with no-map
 properties

Update reserved memory nodes in the AST2700 device tree to include
the no-map property. Add new memory regions for PCIe endpoints,
shared DMA pools, and other subsystems. Adjust existing memory
regions for improved hardware compatibility and feature support.

Introduce conditional definitions for dual-node and PCIe endpoint
configurations, enhancing flexibility in memory configuration.

Change-Id: Ifdfd075ddc5a4523c85f49c2d99e4de20ddd78b3
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit ea7ff8070d8d5021cfd921e2c2251bef0f7e3305)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/dts/ast2700-evb-reserved-mem.dtsi | 70 +++++++++++++++-------
 arch/arm/dts/ast2700-evb.dts               |  3 +
 2 files changed, 51 insertions(+), 22 deletions(-)

diff --git a/arch/arm/dts/ast2700-evb-reserved-mem.dtsi b/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
index dcc5834ab7..d333b18886 100644
--- a/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
+++ b/arch/arm/dts/ast2700-evb-reserved-mem.dtsi
@@ -1,40 +1,71 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
-/* AST2700 reserved memories with fixed address */
-bmc_dev0_memory: bmc-dev0-memory@423800000 {
-	reg = <0x4 0x23800000 0x0 0x100000>;
+/* AST2700 reserved memories with no-map property */
+
+#if PCIE0_EP
+bmc_dev0_memory: bmc-dev0-memory@41b800000 {
+	reg = <0x4 0x1b800000 0x0 0x100000>;
+	no-map;
+};
+
+xdma_memory0: xdma0 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
+	no-map;
+};
+
+pcie0_mmbi0_memory: pcie0-mmbi0-memory@41ba00000 {
+	reg = <0x4 0x1ba00000 0x0 0x200000>;
+	no-map;
+};
+#endif
+
+#if PCIE1_EP
+bmc_dev1_memory: bmc-dev1-memory@41b900000 {
+	reg = <0x4 0x1b900000 0x0 0x100000>;
 	no-map;
 };
 
-bmc_dev1_memory: bmc-dev1-memory@423900000 {
-	reg = <0x4 0x23900000 0x0 0x100000>;
+xdma_memory1: xdma1 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
+	no-map;
+};
+
+pcie1_mmbi4_memory: pcie1-mmbi4-memory@41bc00000 {
+	reg = <0x4 0x1bc00000 0x0 0x200000>;
 	no-map;
 };
+#endif
 
-pcie0_mmbi0_memory: pcie0-mmbi0-memory@423a00000 {
-	reg = <0x4 0x23a00000 0x0 0x200000>;
+pcie2_mmbi0_memory: pcie2-mmbi0-memory@41be00000 {
+	reg = <0x4 0x1be00000 0x0 0x200000>;
 	no-map;
 };
 
-pcie1_mmbi4_memory: pcie1-mmbi4-memory@423c00000 {
-	reg = <0x4 0x23c00000 0x0 0x200000>;
+espi0_mmbi_memory: espi0-mmbi-memory@41c000000 {
+	reg = <0x4 0x1c000000 0x0 0x4000000>;
 	no-map;
 };
 
-pcie2_mmbi0_memory: pcie2-mmbi0-memory@423e00000 {
-	reg = <0x4 0x23e00000 0x0 0x200000>;
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf0: eadf0-memory@420000000 {
+	reg = <0x4 0x20000000 0x0 0x4000000>;
 	no-map;
 };
 
-espi0_mmbi_memory: espi0-mmbi-memory@424000000 {
+#if DUAL_NODE
+espi1_mmbi_memory: espi1-mmbi-memory@424000000 {
 	reg = <0x4 0x24000000 0x0 0x4000000>;
 	no-map;
 };
 
-edaf0: safs-ddr-mode@428000000 {
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf1: edaf1-memory@428000000 {
 	reg = <0x4 0x28000000 0x0 0x4000000>;
 	no-map;
 };
+#endif
 
 ssp_memory: ssp-memory@42c000000 {
 	reg = <0x4 0x2c000000 0x0 0x2000000>;
@@ -46,18 +77,13 @@ tsp_memory: tsp-memory@42e000000 {
 	no-map;
 };
 
-atf: trusted-firmware-a@430000000 {
-	reg = <0x4 0x30000000 0x0 0x80000>;
-	no-map;
-};
-
-optee_core: optee-core@430080000 {
-	reg = <0x4 0x30080000 0x0 0x1000000>;
+ipc_ssp_share: ipc-ssp-share@431080000 {
+	reg = <0x4 0x31080000 0x0 0x800000>;
 	no-map;
 };
 
-ipc_ssp_share: ipc-ssp-share@431080000 {
-	reg = <0x4 0x31080000 0x0 0x800000>;
+ipc_bootmcu_share: ipc-bootmcu-share@431880000 {
+	reg = <0x4 0x31880000 0x0 0x2b0000>;
 	no-map;
 };
 
diff --git a/arch/arm/dts/ast2700-evb.dts b/arch/arm/dts/ast2700-evb.dts
index 4596e307c8..2ce5a5f1aa 100644
--- a/arch/arm/dts/ast2700-evb.dts
+++ b/arch/arm/dts/ast2700-evb.dts
@@ -6,6 +6,9 @@
 #include "ast2700-u-boot.dtsi"
 #include <dt-bindings/net/ti-dp83867.h>
 
+#define DUAL_NODE 0	// 1: DUAL_NODE, 0: SINGLE_NODE
+#define PCIE0_EP 1	// 1: EP, 0: RC
+#define PCIE1_EP 1	// 1: EP, 0: RC
 #define PCIE2_RC 1	// 1: RC, 0: SGMII
 
 / {
-- 
2.34.1

